benchmarks/lobsters/benchmark.rb

```diff
@@ -0,0 +1,44 @@
+require 'harness'
+
+ENV['RAILS_ENV'] ||= 'production'
+ENV['DISABLE_DATABASE_ENVIRONMENT_CHECK'] = '1' # Benchmarks don't really have 'production', so trash it at will.
+
+Dir.chdir __dir__
+use_gemfile extra_setup_cmd: "bin/rails db:drop db:create && sqlite3 db/production.sqlite3 < db/faked_bench_data.sql"
```

Comment:
I'd be fine with this. It'll compress less well in Git, and Maxime seems worried about size. But I think either one would be fine.

Reply:
Are binary sqlite3 files always backward and forward compatible?
Because if not this might error or crash with a newer/older sqlite3 (based on the embedded sqlite3 in the `sqlite3` gem).
Which version of the `sqlite3` package was used to generate this `db/production.sqlite3`? Is it the same as the one in the gem?

IMO it's safer/nicer to just lazily create `db/production.sqlite3` once. But it's true it adds a dependency (unless we could somehow use the gem for this import?)

Reply:
At this point I'm using the gem for the import rather than the command-line utility. So that part's done.

My understanding is that the SQLite DB file format hasn't changed significantly since 2004 (https://www.sqlite.org/fileformat.html). That's not a forward guarantee, obviously.

Yes, the SQLite3 file was generated from a Ruby program, using the gem and the same ActiveRecord models.

It would be possible to lazily generate the db/production.sqlite3 file rather than check it in, definitely. It's pretty slow -- takes several times longer than running the benchmark, as a rule. That's not a problem for most of our use cases (e.g. the big shared benchmarking server) but would suck if you tried to run it on a CI system or similar. It would also need a bit of additional messing around with the database config since we now use a memory database in config/database.yml. But that's entirely doable.

Reply:
Right, that sounds like it should be fine as-is.

One concern is if we ever need to modify this file then this repository would probably explode to 80MB size compressed or so, because AFAIK git is not good at diffing binary stuff.

In any case it would be nice to keep the `db/faked_bench` somewhere just in case it's needed, maybe you could upload it as a gist?

Reply:
The data-generation code, which we got from the original lobsters app, is in lib/tasks/fake_data.rake.

Reply:
At this point I think I'm going to check it in pregenerated, basically because it's hard to guarantee freshness if you're generating from the script. With a configurable script and a checked-in data file, the file *is* the source of truth. But with a configurable script and generation on-demand, you're implying that changing the script should regenerate the file. I'm a little uncomfortable implying that in this case.

Reply:
The SQL commands file is nearly as large as the sqlite3 binary file. Though weirdly, when I delete one and add the other, Git says it's "moved", which makes me wonder. I haven't looked into the binary storage format. But yeah, definitely agree in general that Git is not good at binary diffing.